dj/api/data.py

```diff
@@ -83,3 +87,46 @@ def add_availability(
         status_code=200,
         content={"message": "Availability state successfully posted"},
     )
+
+
+@router.get("/data/{node_name}/")
+def data_for_node(
+    node_name: str,
+    *,
+    dimensions: List[str] = Query([]),
+    filters: List[str] = Query([]),
+    async_: bool = False,
+    session: Session = Depends(get_session),
+    query_service_client: QueryServiceClient = Depends(get_query_service_client),
+) -> QueryWithResults:
+    """
+    Gets data for a node
+    """
+    node = get_node_by_name(session, node_name)
+    if node.type not in (NodeType.METRIC, NodeType.CUBE, NodeType.DIMENSION):
```

Comment:
@shangyian I am trying to come up with a good reason for not allowing users to query data for a source or transform node but I can't. We can always add a default row limit, on any node type. Just trying to understand this restriction better.

Reply:
I guess I just feel like it's a bit unusual for people to try to access sources or transforms through DJ, but I can also see how if someone is in the system and exploring, it might be useful. But in my view the most powerful part of DJ is its ability to generate queries, and for source nodes that's not going to be applicable. I could see being used for transform nodes though.

Reply:
> But in my view the most powerful part of DJ is its ability to generate queries

Fully agree ^, but in this case I think we don't need to tell people that they can't see data for a source or transform node if they just wanna have a peek at `SELECT * FROM any_node LIMIT 10` or something. It just feels like an artificial restriction, even if this is not a common use case.

Reply:
Makes sense! I just added some changes to support querying both source and transform nodes. It turns out the existing `get_query` doesn't support nodes with no queries, so I added some modifications to generate a select star query AST.